### PR TITLE
rename mitmproxy.filt -> mitmproxy.flowfilter

### PR DESCRIPTION
**Pull request by Kriechi**
There seems to be a bug here, which is now fixed:
https://github.com/mitmproxy/mitmproxy/blob/d068000f16cb44b7277de04ba20bd1fc62e4a956/mitmproxy/builtins/filestreamer.py#L30

Also I removed a bit of code duplication here:
https://github.com/mitmproxy/mitmproxy/pull/1587/files#diff-6aaebdf9a233f01096f39e9db9f3d091L63

Since `flow.match` will be removed, I added a `DeprecationWarning` and moved the actual function to `flowfilter.match`.

### CODE REVIEW

**Comment by Kriechi** [comment]
meh - lazy Python excuse IMHO :D
`filt` is just super weird - should we rename the import to e.g. `flow_filter` or something?

Either way, there is a bug in the above linked code, that needs fixing.

**Comment by cortesi** [comment]
Max is right - I used `filt` because `filter` is a builtin. Some of the most infuriating Python bugs I've ever had to track down have been due to builtin name clashes (cue Max with "but if you used PyCharm..."). 

I agree that `filt` is, as you say, super weird and should be changed, but I also don't like `flow_filter`, which seems verbose. Maybe some sort of half-hearted compromise like "filtr"? Any other suggestions?

**Comment by Kriechi** [comment]
We are not trying to make a hipster web-startup company... `filtr`... *SCNR* 😄 
Other ideas are welcome!

**Comment by Kriechi** [comment]
@mhils @cortesi: I updated this PR according to our Slack discussion yesterday.

File: mitmproxy/flowfilter.py [hunk 1]
```diff
@@ -492,25 +493,42 @@ def _make():
...
+        filter = bnf.parseString(s, parseAll=True)[0]
+        filter.pattern = s
+        return filter
     except pp.ParseException:
         return None
     except ValueError:
         return None
 
 
+def match(flow, flt):
```
Comment (mhils) [inline mitmproxy/flowfilter.py]: I would prefer a `match(flt, flow):` signature, as that would be in line with `re.match`. Thoughts?

File: mitmproxy/flowfilter.py [hunk 2]
```diff
@@ -492,25 +493,42 @@ def _make():
 def parse(s):
     # type: (str) -> TFilter
     try:
-        filt = bnf.parseString(s, parseAll=True)[0]
-        filt.pattern = s
-        return filt
+        filter = bnf.parseString(s, parseAll=True)[0]
```
Comment (mhils) [inline mitmproxy/flowfilter.py]: As tempting as it is, we don't want to override builtins :wink: 
filtexpr, filter_function, flowfilter, ... ?

File: mitmproxy/flowfilter.py (hunk 1 above)
Comment (Kriechi) [inline mitmproxy/flowfilter.py]: Makes perfect sense! Done.

File: mitmproxy/flowfilter.py (hunk 2 above)
Comment (Kriechi) [inline mitmproxy/flowfilter.py]: meh - I fixed it.

**Pull request merged**